tools/c7n_gcp/c7n_gcp/policy.py

```diff
@@ -101,8 +103,47 @@ def run(self, event, context):
         return PullMode.run(self)
 
 
+class FunctionModeImpl(FunctionMode):
+    def resolve_resources(self, event):
+        raise NotImplementedError("subclass responsibility")  # pragma: no cover
```

Comment:
If we're going to put this abstract method definition somewhere, seems like it could make more sense higher up (in [ServerlessExecutionMode](https://github.com/cloud-custodian/cloud-custodian/blob/d228a985b604b191ded1b98567bae710f1803c16/c7n/policy.py#L289-L300) for example) since there are existing `resolve_resource()` implementations across providers.

Reply:
I was trying to keep this isolated to `c7n_gcp`. I have future plans for more reuse of base types when we have the c7n.core.

Reply:
Gotcha, fair point